include/seastar/core/sstring.hh

```diff
@@ -346,6 +346,29 @@ public:
         return *this;
     }
 
+    /**
+     * Append the given character to the end of the string
+     * @param ch  The character to append
+     * @note unlike @c std::basic_string, sstring does not preallocate, so
+     *       this call always lead to reallocation if external storage is
+     *       used.
+     */
+    constexpr void push_back(char_type ch) {
+        if (is_internal()) {
+            if (static_cast<Size>(u.internal.size) < max_size) {
+                u.internal.str[u.internal.size++] = ch;
+                if (NulTerminate) {
+                    u.internal.str[u.internal.size] = '\0';
+                }
+                return;
+            }
+        }
+        basic_sstring new_str(initialized_later(), u.external.size + 1);
```

Comment:
What if we reached here despite is_internal() (because we reached max size)? Isn't u.external.size wrong?
Please make sure you tests check the case of an empty string growing and growing until passing the internal size and see it still works.